annotate.el

```diff
@@ -896,5 +922,48 @@ file."
   (with-temp-file annotate-file
     (prin1 data (current-buffer))))
 
+(define-button-type 'annotate-summary-button
+  'follow-link t
+  'help-echo "Click to show")
+
+(defun annotate-summary-button-pressed (button)
+  "Callback called when an annotate-summary-button is activated"
+  (let ((buffer (find-file-other-window (button-get button 'file))))
+    (with-current-buffer buffer
+      (goto-char (button-get button 'go-to)))))
+
+(defun annotate-show-annotation-summary ()
+  "Show a summary of all the annotations in a temp buffer"
+  (interactive)
+  (cl-labels ((ellipsize (text)
+                         (if (> (string-width text)
+                                annotate-summary-link-max-width)
+                             (concat (subseq text 0 (- annotate-summary-link-max-width 3))
+                                     "...")
+                           text)))
+    (with-temp-buffer-window
+     "*annotations*" nil nil
+     (with-current-buffer "*annotations*"
```

Comment:
Are both `with-temp-buffer-window` and `with-current-buffer` necessary, or would one of them suffice?

Reply:
I think `with-temp-buffer-window` is needed so I tried to remove `with-current-buffer` and the summary text was inserted in the selected buffer instead of the summary window just created (i mean, the one that should hosts the summary), in my case the text was inserted straight in the source code, that was creepy! :)

Reply:
OK, interesting.

Reply:
well, seems that with a couple of expression more i was able to get rid of the second macro (`with-current-buffer`)!